### PR TITLE
Irfinder

### PR DESCRIPTION
**Pull request by khourhin**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

First contribution
Adding recipe for [IRFinder](https://github.com/williamritchie/IRFinder)

### CODE REVIEW

File: recipes/irfinder/build.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+cd $SRC_DIR/src/irfinder && make && mv irfinder $SRC_DIR/bin
+cd $SRC_DIR/src/trim && make && mv trim $SRC_DIR/bin
+cd $SRC_DIR/src/winflat && make && mv winflat $SRC_DIR/bin
+
+cd $PREFIX
+
+irfinder_home=$PREFIX/opt/irfinder-$PKG_VERSION
```
Comment (bgruening) [inline recipes/irfinder/build.sh]: I think a simple `cp $SRC_DIR/bin/* $PREFIX/bin/` should be enough.

File: recipes/irfinder/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  number: 0
+  
+source:
+  git_rev: v1.2.2
+  url: https://github.com/williamritchie/IRFinder/archive/v1.2.2.tar.gz
+  md5: bf5d672b7ff61feaf24d1299b87fa2f1
+  
+requirements:
+  build:
+    - gcc
```
Comment (bgruening) [inline recipes/irfinder/meta.yaml]: please read here about comiled packages: https://bioconda.github.io/guidelines.html#c-c

File: recipes/irfinder/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  run:
+    - perl
+    - r-base
+      
+test:
+  commands:
+    - 'IRFinder -v &> /dev/null'
```
Comment (bgruening) [inline recipes/irfinder/meta.yaml]: You could test a few more packages.
`&> /dev/null` is not needed.

File: recipes/irfinder/build.sh (hunk 1 above)
Comment (khourhin) [inline recipes/irfinder/build.sh]: I'm not sure, 

IRFinder is apparently using a perl wrapper script which execute scripts which are in bin/util. That's why I went for the option of putting the whole directory (with bin and utils) in /opt. I could also limit to the bin directory ie:

cp -r $SRC_DIR/bin $irfinder_home

Let me know if I miss a better solution which will not break the dependencies neither clutter the bin directory with them.

IRFinder bin tree:

bin/
├── AdaptorDetect.pl
├── analysisWithLowReplicates.pl
├── analysisWithNoReplicates.pl
├── DESeq2Constructor.R
├── IRFinder
└── util/
    ├── bed-to-intron+exon.pl
    ├── Build-BED-refs.sh
    ├── generateReadsError.pl
    ├── gtf2bed-custom.pl
    ├── IntronExclusion.pl
    ├── irfinder
    ├── IRFinder-BuildRefFromEnsembl
    ├── Mapability
    ├── trim
    ├── warnings
    └── winflat

File: recipes/irfinder/meta.yaml (hunk 2 above)
Comment (khourhin) [inline recipes/irfinder/meta.yaml]: Indeed I missed this one. I'm relatively new to colloboration with github. What would be the best/cleanest way to make the changes (ie change on my local branch, push to my account, do a new pull request, or make the changes directly on github) ?

File: recipes/irfinder/meta.yaml (hunk 3 above)
Comment (khourhin) [inline recipes/irfinder/meta.yaml]: True, I'm going to check this

Comment (khourhin) [inline recipes/irfinder/meta.yaml]: Mmm, after checking actually I'm not sure what could be added here. Few information is available in the manual and article about the other scripts present in bin/. I don't see here what informative test could be done which would not involve data test (which we are not supposed to insert here right ?).

**Comment by bgruening** [comment]
Thanks @khourhin!

File: recipes/irfinder/meta.yaml (hunk 2 above)
Comment (bgruening) [inline recipes/irfinder/meta.yaml]: :)

**Comment by khourhin** [comment]
Thanks for the help !

**Pull request merged**